src/Magento/FunctionalTestingFramework/Util/ModuleResolver.php

```diff
@@ -314,46 +429,193 @@ private static function globRelevantWrapper($testPath, $pattern)
     }
 
     /**
-     * Takes a multidimensional array of module paths and flattens to return a one dimensional array of test paths
+     * Aggregate all code paths with test module composer json files
      *
-     * @param array $modulePaths
      * @return array
      */
-    private function flattenAllModulePaths($modulePaths)
+    private function aggregateTestModulePathsFromComposerJson()
     {
-        $it = new \RecursiveIteratorIterator(new \RecursiveArrayIterator($modulePaths));
-        $resultArray = [];
+        // Define the module paths
+        $magentoBaseCodePath = MAGENTO_BP;
 
-        foreach ($it as $value) {
-            $resultArray[] = $value;
+        // Define the module paths from default TESTS_MODULE_PATH
+        $modulePath = defined('TESTS_MODULE_PATH') ? TESTS_MODULE_PATH : TESTS_BP;
+        $modulePath = rtrim($modulePath, DIRECTORY_SEPARATOR);
+
+        $searchCodePaths = [
+            $magentoBaseCodePath . DIRECTORY_SEPARATOR . self::DEV_TESTS,
+        ];
+
+        // Add TESTS_MODULE_PATH if it's not included
+        if (array_search($modulePath, $searchCodePaths) === false) {
+            $searchCodePaths[] = $modulePath;
         }
 
-        return $resultArray;
+        return $this->getComposerJsonTestModulePaths($searchCodePaths);
     }
 
     /**
-     * Runs through enabled modules and maps them known module paths by name.
-     * @param array $enabledModules
-     * @param array $allModulePaths
+     * Retrieve all module code paths that have test module composer json files
+     *
+     * @param array $codePaths
      * @return array
      */
-    private function getEnabledDirectoryPaths($enabledModules, $allModulePaths)
+    private function getComposerJsonTestModulePaths($codePaths)
     {
-        $enabledDirectoryPaths = [];
-        foreach ($enabledModules as $magentoModuleName) {
-            if (!isset($this->knownDirectories[$magentoModuleName]) && !isset($allModulePaths[$magentoModuleName])) {
-                continue;
-            } elseif (isset($this->knownDirectories[$magentoModuleName])
-                && !isset($allModulePaths[$magentoModuleName])) {
-                LoggingUtil::getInstance()->getLogger(ModuleResolver::class)->warn(
-                    "Known directory could not match to an existing path.",
-                    ['knownDirectory' => $magentoModuleName]
-                );
+        if (null !== $this->composerJsonModulePaths) {
+            return $this->composerJsonModulePaths;
+        }
+        try {
```

Comment:
getTestModules() could throw TestFrameworkException when directory passed in is invalid. It will be in logs but here we ignore this exception and continue execution.

Reply:
Gotcha.